### ISSUE TITLE
Why am I getting accuracy 0.0?

### ISSUE THREAD

**Issue by nairvinitha** [nairvinitha/ResNet50-with-Random-Forest-Classifier#2]
I am running the following code on Google Colab:
import numpy as np 
import matplotlib.pyplot as plt
import glob
import cv2

from keras.models import Model, Sequential
from keras.layers import Dense, Flatten, Conv2D, MaxPooling2D
from tensorflow.keras.layers import BatchNormalization
import os
import seaborn as sns
import tensorflow as tf
from tensorflow.keras.applications import ResNet50

print(os.listdir(r"C:\Users\vinit\Desktop\EHU\Semester 3 - RnD project\Organic Waste"))

SIZE = 128

train_images = []
train_labels = [] 

for directory_path in glob.glob(r"C:\Users\vinit\Desktop\EHU\Semester 3 - RnD project\Organic Waste\train\*"):
    label = directory_path.split("\\")[-1]
    print(label)
    for img_path in glob.glob(os.path.join(directory_path, "*.jpg")):
        print(img_path)
        img = cv2.imread(img_path, cv2.IMREAD_COLOR)
        print(img)
        try:
            img = cv2.resize(img, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img.shape)
        except:
            break
        height, width , layers = img.shape
        size=(width,height)
        print(size)
        #img = cv2.resize(img, (SIZE, SIZE))
        img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
        train_images.append(img)
        train_labels.append(label)
        
    for img_path1 in glob.glob(os.path.join(directory_path, "*.jpeg")):
        print(img_path1)
        img1 = cv2.imread(img_path1, cv2.IMREAD_COLOR)
        print(img1)
        try:
            img1 = cv2.resize(img1, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img1.shape)
        except:
            break
        height1, width1 , layers1 = img1.shape
        size1=(width1,height1)
        print(size1)
        #img = cv2.resize(img, (SIZE, SIZE),fx=0,fy=0, interpolation = cv2.INTER_CUBIC)
        img1 = cv2.cvtColor(img1, cv2.COLOR_RGB2BGR)
        train_images.append(img1)
        train_labels.append(label)
        
        
    for img_path2 in glob.glob(os.path.join(directory_path, "*.png")):
        print(img_path2)
        img2 = cv2.imread(img_path2, cv2.IMREAD_COLOR)
        print(img2)
        try:
            img2 = cv2.resize(img2, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img2.shape)
        except:
            break
        height2, width2 , layers2 = img2.shape
        size2=(width2,height2)
        print(size2)
        #img = cv2.resize(img, (SIZE, SIZE),fx=0,fy=0, interpolation = cv2.INTER_CUBIC)
        img2 = cv2.cvtColor(img2, cv2.COLOR_RGB2BGR)
        train_images.append(img2)
        train_labels.append(label)
        
        
    for img_path3 in glob.glob(os.path.join(directory_path, "*.WEBP")):
        print(img_path3)
        img3 = cv2.imread(img_path3, cv2.IMREAD_COLOR)
        print(img3)
        try:
            img3 = cv2.resize(img3, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img3.shape)
        except:
            break
        height3, width3 , layers3 = img3.shape
        size3=(width,height)
        print(size3)
        #img = cv2.resize(img, (SIZE, SIZE),fx=0,fy=0, interpolation = cv2.INTER_CUBIC)
        img3 = cv2.cvtColor(img3, cv2.COLOR_RGB2BGR)
        train_images.append(img3)
        train_labels.append(label)

        
train_images = np.array(train_images)
train_labels = np.array(train_labels)
len(train_images)

# test
test_images = []
test_labels = [] 

for directory_path in glob.glob(r"C:\Users\vinit\Desktop\EHU\Semester 3 - RnD project\Organic Waste\test\*"):
    label1 = directory_path.split("\\")[-1]
    print(label1)
    for img_path in glob.glob(os.path.join(directory_path, "*.jpg")):
        print(img_path)
        img = cv2.imread(img_path, cv2.IMREAD_COLOR)
        print(img)
        try:
            img = cv2.resize(img, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img.shape)
        except:
            break
        height, width , layers = img.shape
        size=(width,height)
        print(size)
        #img = cv2.resize(img, (SIZE, SIZE))
        img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
        test_images.append(img)
        test_labels.append(label1)
        
    for img_path1 in glob.glob(os.path.join(directory_path, "*.jpeg")):
        print(img_path1)
        img1 = cv2.imread(img_path1, cv2.IMREAD_COLOR)
        print(img1)
        try:
            img1 = cv2.resize(img1, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img1.shape)
        except:
            break
        height1, width1 , layers1 = img1.shape
        size1=(width1,height1)
        print(size1)
        #img = cv2.resize(img, (SIZE, SIZE),fx=0,fy=0, interpolation = cv2.INTER_CUBIC)
        img1 = cv2.cvtColor(img1, cv2.COLOR_RGB2BGR)
        test_images.append(img1)
        test_labels.append(label1)
        
    for img_path2 in glob.glob(os.path.join(directory_path, "*.png")):
        print(img_path2)
        img2 = cv2.imread(img_path2, cv2.IMREAD_COLOR)
        print(img2)
        try:
            img2 = cv2.resize(img2, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img2.shape)
        except:
            break
        height2, width2 , layers2 = img2.shape
        size2=(width2,height2)
        print(size2)
        #img = cv2.resize(img, (SIZE, SIZE),fx=0,fy=0, interpolation = cv2.INTER_CUBIC)
        img2 = cv2.cvtColor(img2, cv2.COLOR_RGB2BGR)
        test_images.append(img2)
        test_labels.append(label1)
        
    for img_path3 in glob.glob(os.path.join(directory_path, "*.WEBP")):
        print(img_path3)
        img3 = cv2.imread(img_path3, cv2.IMREAD_COLOR)
        print(img3)
        try:
            img3 = cv2.resize(img3, (SIZE, SIZE), interpolation=cv2.INTER_AREA)
            print(img3.shape)
        except:
            break
        height3, width3 , layers3 = img3.shape
        size3=(width3,height3)
        print(size3)
        #img = cv2.resize(img, (SIZE, SIZE),fx=0,fy=0, interpolation = cv2.INTER_CUBIC)
        img3 = cv2.cvtColor(img3, cv2.COLOR_RGB2BGR)
        test_images.append(img3)
        test_labels.append(label1)
        
        
test_images = np.array(test_images)
test_labels = np.array(test_labels)
len(test_images)

#Encode labels from text to integers.
from sklearn import preprocessing
le = preprocessing.LabelEncoder()
le.fit(test_labels)
test_labels_encoded = le.transform(test_labels)
le.fit(train_labels)
train_labels_encoded = le.transform(train_labels)

#Split data into test and train datasets (already split but assigning to meaningful convention)
x_train, y_train, x_test, y_test = train_images, train_labels_encoded, test_images, test_labels_encoded

# Normalize pixel values to between 0 and 1
x_train, x_test = x_train / 255.0, x_test / 255.0

resnetmodel = ResNet50(include_top=False,
    weights="imagenet",
    input_shape= (SIZE,SIZE,3)
)
for layers in resnetmodel.layers:
    layers.trainable = False
resnetmodel.summary()

#using the features of the ResNet model for Random Forest classifier
extract_features = resnetmodel.predict(x_train)

feature_labels = extract_features.reshape(extract_features.shape[0],-1)
feature_labels

RF_input = feature_labels
#using Random Forest as a classifier
from sklearn.ensemble import RandomForestClassifier
RandomForest = RandomForestClassifier(n_estimators= 1000, random_state=42)

#Train the Random Forest model on the training dataset
RandomForest.fit(RF_input,y_train)

#follow the same process for the test dataset
extractfeatures_test = resnetmodel.predict(x_test)

featurelabels_test = extractfeatures_test.reshape(extractfeatures_test.shape[0],-1)

#Predict the output using the Random Forest Model
RF_predict = RandomForest.predict(featurelabels_test)

#Inverse 'le' to get the original label back
#RF_predict = le.inverse_transform(RF_predict)
#RF_predict.shape

#Get the overall accuracy
from sklearn import metrics
print("Accuracy = ", metrics.accuracy_score(test_labels, RF_predict))

Output: 
Accuracy =  0.0
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:217: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  score = y_true == y_pred

Why does this happen? How do I fix it? Please help.
